components/tensorflow-notebook-image/start-notebook.sh

```diff
@@ -15,9 +15,5 @@
 
 set -e
 
-if [[ ! -z "${JUPYTERHUB_API_TOKEN}" ]]; then
-  # launched by JupyterHub, use single-user entrypoint
-  exec /usr/local/bin/start-singleuser.sh $*
-else
-  . /usr/local/bin/start.sh jupyter notebook $*
-fi
+jupyter lab --notebook-dir=/home/jovyan --ip=0.0.0.0 --no-browser --allow-root --port=8888
```

Comment:
Is this starting the jupyter lab interface by default? I think we decided we wanted to stick with Jupyter by default.

Reply:
I understand why you want to have the classic notebook interface be the default, and I agree.
It would be great if we can have a way for the user to choose.
Based on the discussion at https://github.com/kubeflow/kubeflow/pull/2463#pullrequestreview-204068952, you preferred an approach where we didn't have a Jupyter-specific section in the Notebook CRD.
Since we agree that the best way is to have a pre-baked default inside the images, for the type of environment to use, can we have two sets of images being built and published by Kubeflow automatically? One built with classic Jupyter notebook being the default, one built with JupyterLab being the default, so users can switch easily without having to rebuild images?

The proposed image will be the one with the classic interface, so the current TFX widgets work, but having an alternative readily available lifts a major obstacle to working with JupyterLab, having to rebuild the images.

Reply:
@vkoukis Building multiple images seems like a heavy hammer. One of my hopes is that we can make it easier to reuse any jupyter image see #2208. With that in mind when we do publish jupyter images we should do that in the context of some specific use case and make choices suitable for that application.

For example, I think it might make sense to have a TFX focused Jupyter image that would have the TFX libraries installed and make Jupyter the default for that reason.

Reply:
@jlewi just saw this, I'd missed it.
> Building multiple images seems like a heavy hammer. One of my hopes is that we can make it easier to reuse any jupyter image see #2208.
I agree with this in general. I think Kubeflow can focus only on building Images that contain Kubeflow-specific components, e.g., Fairing.

On whether this is heavy handed, I'll experiment a bit with building different images starting from existing ones as a base. so we can minimize the size of extra layers. Changing an entry point will be essentially free, both in build and download time, but can prove beneficial.

>For example, I think it might make sense to have a TFX focused Jupyter image that would have the TFX libraries installed and make Jupyter the default for that reason.

Agreed.
I'll experiment a bit and come back with a PR for discussion. Thanks!